spring-cloud-gcp-starters/spring-cloud-gcp-starter-config/src/main/java/org/springframework/cloud/gcp/config/GoogleConfigPropertySourceLocator.java

```diff
@@ -0,0 +1,138 @@
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.config;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.auth.Credentials;
+
+import org.springframework.cloud.bootstrap.config.PropertySourceLocator;
+import org.springframework.cloud.config.client.ConfigClientProperties;
+import org.springframework.cloud.config.client.ConfigServicePropertySourceLocator;
+import org.springframework.cloud.gcp.core.GcpProjectIdProvider;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.core.env.PropertySource;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.SimpleClientHttpRequestFactory;
+import org.springframework.util.Assert;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Custom property source locator for Google Cloud Runtime Configurator.
+ *
+ * @author Jisha Abubaker
+ */
+@Configuration
+@ComponentScan("org.springframework.cloud.gcp.core")
+public class GoogleConfigPropertySourceLocator implements PropertySourceLocator {
+
+	private static final String RUNTIMECONFIG_API_ROOT = "https://runtimeconfig.googleapis.com/v1beta1/";
+
+	private static final String ALL_VARIABLES_PATH =
+			"projects/{project}/configs/{name}_{profile}/variables?returnValues=true";
+
+	private static final String PROPERTY_SOURCE_NAME = "spring-cloud-gcp";
+
+	private static final String AUTHORIZATION_HEADER = "Authorization";
+
+	private static final String BEARER_TOKEN_PREFIX = "Bearer ";
+
+	private ConfigClientProperties defaultProperties;
+
+	private Credentials credentials;
+
+	private String projectId;
+
+	public GoogleConfigPropertySourceLocator(CredentialsProvider credentialsProvider,
+			GcpProjectIdProvider projectIdProvider, ConfigClientProperties defaultProperties) throws Exception {
+		Assert.notNull(credentialsProvider, "Credentials provider cannot be null");
+		Assert.notNull(projectIdProvider, "Project ID provider cannot be null");
+		Assert.notNull(defaultProperties, "Config client properties cannot be null");
+		this.credentials = credentialsProvider.getCredentials();
+		this.projectId = projectIdProvider.getProjectId();
```

Comment:
Should we check that the returned `credentials` and `projectId` are also not null?

Reply:
done.